File: 09-12-2024_Insert_Interval.py
class Solution:

    def insertInterval(self, intervals, newInterval):
        ans = []
        i = 0
        n = len(intervals)

        while i < n and newInterval[0] > intervals[i][1]:
            ans.append(intervals[i])
            i += 1


        while i < n and intervals[i][0] <= newInterval[1]:
            newInterval[0] = min(intervals[i][0], newInterval[0])
            newInterval[1] = max(intervals[i][1], newInterval[1])
            i += 1
        ans.append(newInterval)


        while i < n:
            ans.append(intervals[i])
            i += 1
        return ans

File: 09-12-2024_Insert_Interval.cpp
class Solution {
  public:
    vector<vector<int>> insertInterval(vector<vector<int>> &intervals,
                                       vector<int> &newInterval) {
        intervals.push_back(newInterval);
        sort(intervals.begin(), intervals.end());
        
        int n = intervals.size();
        int i = 0;
        vector<vector<int>>ans;
        while(i < n){
            int start = intervals[i][0], end = intervals[i][1];
            while(i+1 < n && end >= intervals[i+1][0]){
                end = max(end, intervals[i+1][1]);
                i++;
            }
            ans.push_back({start, end});
            i++;
        }
        return ans;
    }
};
